igvm/hypervisor.py

```diff
@@ -555,9 +556,23 @@ def vm_block_device_name(self):
         the guest OS"""
         return 'vda1'
 
+    def migrate_wait_shutdown(
+        self, vm: VM, no_shutdown: bool, transaction: Transaction,
+    ):
+        vm.set_state('maintenance', transaction=transaction)
+        if vm.is_running():
+            if no_shutdown:
+                log.info('Please shut down the VM manually now')
+                vm.wait_for_running(running=False, timeout=86400)
+            else:
+                vm.shutdown(
+                    check_vm_up_on_transaction=False,
+                    transaction=transaction,
+                )
+
     def migrate_vm(
-        self, vm, target_hypervisor, offline, offline_transport, transaction,
-        no_shutdown,
+        self, vm: VM, target_hypervisor: 'Hypervisor', offline: bool,
+        offline_transport: str, transaction: Transaction, no_shutdown: bool,
```

Comment:
You can make adding type hints a dedicated commit the next time to make it easier for people to distinct style changes from actual changes.